lib/LANraragi/Controller/Api/Archive.pm

```diff
@@ -239,41 +256,47 @@ sub create_archive {
         $thumbhash = $redis->hget( $id, "thumbhash" );
         $thumbhash = LANraragi::Utils::Database::redis_decode($thumbhash);
     }
+    $redis->del("upload:$filename");
     $redis->quit();
 
-    # modify status based on handler's return message.
+    # handle all post-upload errors, setting a server-side error status by default.
     if ( $success_status==0 ) {
         $status = 500;
-        if ( index($message, "Unsupported File Extension") != -1 ) {
-            $status = 415;
-        } elsif ( index($message, "Enable replace duplicated archive in config to replace old ones") != -1 ) {
-            $status = 409;
-        } elsif ( index($message, "The file couldn't be moved to your content folder") != -1 ) {
-            $status = 500;
+        my $error = "Server Error";
+        if ( index($message, "Enable replace duplicated archive in config to replace old ones") != -1 ) {
```

Comment:
Doing a string comparison from the user-facing message of `handle_incoming_file` feels very brittle.. Maybe we should have different values for `success_status` instead? Like 1, 2, -1.. Or maybe even http error codes at this point.

Reply:
Alternatively I guess we could just return 400 for all errors and rely on clients to look at `message` for details -- That's how other endpoints in the server do it.

Reply:
Yeahhhh I can't think of a really good way to do this, this was just one implementation I thought of. I'm tending towards returning all 400, but that mixes in server-side errors and passes the brittleness over to the client-side.

An option that removes the need for string comparison is to copy the logic of `handle_incoming_file` to the controller level and inject response handling there, which I'm fine doing but might look ugly. Or refactor and break apart `handle_incoming_file` so that it can be used by the controller.

Adding an http status code return to `handle_incoming_file` can work in a backwards-compatible way. Otherwise I don't have a super strong opinion on these approaches.

Reply:
decided to fix this by replacing `handle_incoming_file` with http status codes, and doing status code => success_status mapping for minion downstream

Reply:
Yep, that's perfectly fine for me. We'll probably want to mention those various status code in the documentation.